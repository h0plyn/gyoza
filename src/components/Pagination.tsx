import React, { FC, Fragment } from 'react';
import { Page } from '../types';

const Pagination: FC<Page> = ({ page, setPage }: Page) => {
  const coinsPerPage = 10;
  const totalPages = 100 / coinsPerPage;
  const pagination = new Array(totalPages).fill(0);

  return (
    <div style={{ marginBottom: '3rem' }}>
      <div className="pagination" title="pagination">
        {pagination.map((_, idx) => {
          const pageNumber = idx + 1;
          const currentPage = page === pageNumber;
          return (
            <div
              key={pageNumber}
              className="pagination-numbers"
              id={currentPage ? 'current-page' : ''}
              onClick={() => setPage(pageNumber)}
            >
              {pageNumber}
            </div>
          );
        })}
      </div>
      <button
        disabled={page > totalPages - 1 ? true : false}
        onClick={() => setPage(page + 1)}
      >
        Next
      </button>
    </div>
  );
};

export default Pagination;
