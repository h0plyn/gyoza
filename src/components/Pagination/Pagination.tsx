import { FC } from 'react';
import styled from 'styled-components';
import { Page } from '../../types';

const PaginationStyles = styled.div`
  margin-bottom: 3rem;

  & .pagination {
    display: flex;
    padding: 1rem;
  }

  & .pagination-numbers:not(:first-child) {
    margin-left: 0.5rem;
  }

  & #current-page {
    color: var(--tertiary);
  }
`;

const Pagination: FC<Page> = ({ page, setPage }: Page) => {
  const coinsPerPage = 10;
  const totalPages = 100 / coinsPerPage;
  const pagination = new Array(totalPages).fill(0);

  return (
    <PaginationStyles>
      <div className="pagination" title="pagination" data-testid="pagination">
        {pagination.map((_, idx) => {
          const pageNumber = idx + 1;
          const currentPage = page === pageNumber;
          return (
            <div
              key={pageNumber}
              className="pagination-numbers"
              id={currentPage ? 'current-page' : ''}
              onClick={() => setPage(pageNumber)}
              data-testid="page"
            >
              {pageNumber}
            </div>
          );
        })}
      </div>
      <button
        disabled={page > totalPages - 1 ? true : false}
        onClick={() => setPage(page + 1)}
      >
        Next
      </button>
    </PaginationStyles>
  );
};

export default Pagination;
