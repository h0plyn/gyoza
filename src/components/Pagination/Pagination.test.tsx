import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';

describe('<Pagination />', () => {
  afterEach(() => {
    cleanup();
  });

  const setup = () => {
    let page = 1;
    const setPage = jest.fn();
    const utils = render(<Pagination page={page} setPage={setPage} />);
    const button = utils.getByRole('button');
    const pages = utils.getAllByTestId('page');
    return { page, setPage, pages, button, ...utils };
  };

  it('should render the Pagination component without crashing', () => {
    render(<Pagination />);
  });

  it('matches the snapshot of Pagination', () => {
    const tree = renderer.create(<Pagination />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a button', () => {
    const { button } = setup();
    expect(button).toBeDefined();
  });

  it('should display numbers representing 10 pages of coins', () => {
    const { pages } = setup();
    expect(pages.length).toBe(10);
    expect(screen.queryByText('0')).not.toBeInTheDocument();
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('10')).toBeInTheDocument();
    expect(screen.queryByText('11')).not.toBeInTheDocument();
  });

  it('the button should fire a click handler', () => {
    const { button, setPage } = setup();
    fireEvent.click(button);
    expect(setPage).toHaveBeenCalled();
  });
});
