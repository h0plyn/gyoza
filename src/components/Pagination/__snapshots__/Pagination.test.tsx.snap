// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches the snapshot of Pagination 1`] = `
<div
  className="sc-bdnxRM gdwPWC"
>
  <div
    className="pagination"
    data-testid="pagination"
    title="pagination"
  >
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="pagination-numbers"
      data-testid="page"
      id=""
      onClick={[Function]}
    >
      10
    </div>
  </div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
